src/puppetlabs/puppetdb/import.clj

```diff
@@ -4,19 +4,44 @@
    This is a command-line tool for importing data into PuppetDB. It expects
    as input a tarball generated by the PuppetDB `export` command-line tool."
   (:import [org.apache.commons.compress.archivers.tar TarArchiveEntry]
-           [puppetlabs.puppetdb.archive TarGzReader])
+           [puppetlabs.puppetdb.archive TarGzReader]
+           [java.io File])
   (:require [clojure.java.io :as io]
             [clojure.tools.logging :as log]
             [puppetlabs.puppetdb.archive :as archive]
             [puppetlabs.puppetdb.reports :as reports]
             [puppetlabs.puppetdb.export :as export]
             [puppetlabs.puppetdb.utils :as utils]
             [puppetlabs.puppetdb.cheshire :as json]
-            [puppetlabs.puppetdb.schema :refer [defn-validated]]))
+            [puppetlabs.puppetdb.schema :refer [defn-validated]]
+            [puppetlabs.puppetdb.utils :as utils]))
 
-(defn file-pattern
-  [entity]
-  (re-pattern (str "^" (.getPath (io/file utils/export-root-dir entity ".*\\.json")) "$")))
+(defn make-certname-matcher
+  "Returns a function that find a certname in the `file-path` for the
+  given `entity`, the function returns nil if the path is not
+  recognized"
+  [entity with-hash?]
+  (let [command-certname-regex (re-pattern
+                                (str "^"
+                                     (apply str
+                                            (map utils/regex-quote
+                                                 [utils/export-root-dir
+                                                  File/separatorChar
```

Comment:
If I'm reading the broader context right, I wondered if we might not want th current platform separator here (i.e. if this is matching tar entries);  I'm not positive yet, but it's looking like the separator there might "always" be "/".